components/include/cn_memory.h

```diff
@@ -69,4 +69,31 @@ ensures
     take i = each(u64 j; j >= 0u64 && j < n) {Block<uint8_t>(array_shift<uint8_t>(return, j))};
 $*/
 
+/*$
+datatype OptionMemory {
```

Comment:
I would like to have at least a sentence of description for each predicate, similar to what Gus did for [array_utils.h](https://github.com/GaloisInc/VERSE-OpenSUT/blob/main/components/include/cn_array_utils.h). I know this might be obvious to you, but this will really pay off during the transition. Maybe @peterohanley can help here?